pkg/v1/providers/tests/unit/ip_family_test.go

```diff
@@ -709,6 +708,55 @@ var _ = Describe("TKG_IP_FAMILY Ytt Templating", func() {
 					Expect(output).To(HaveYAMLPathWithValue(ipFamilyPath, "ipv6,ipv4"))
 				})
 			})
+			Context("exclude the vsphere control plane endpoint from node ip selection", func() {
+				var excludeInternalNetworkSubnetCidr = "$.data.vsphereCPI.vmExcludeInternalNetworkSubnetCidr"
+				var excludeExternalNetworkSubnetCidr = "$.data.vsphereCPI.vmExcludeExternalNetworkSubnetCidr"
+				When("VSPHERE_CONTROL_PLANE_ENDPOINT is ipv4", func() {
+					It("excludes it as a CIDR from both external and internal node ip selection", func() {
+						values := createDataValues(map[string]string{
+							"PROVIDER_TYPE":                  "vsphere",
+							"TKG_IP_FAMILY":                  "ipv4",
```

Comment:
A bit interested what happens if given "TKG_IP_FAMILY: ipv6" but provide a ipv4 address for VSPHERE_CONTROL_PLANE_ENDPOINT.

Reply:
That scenario is handled by this validation code: 
https://github.com/vmware-tanzu/tanzu-framework/blob/d2680d50ba976761e8f407516618d78253350859/pkg/v1/tkg/client/validate.go#L1778-L1802

Reply:
oh, wait, I'm mistaken, that does not validate the scenario you've described...

Reply:
We realize that we don't know what happens in that scenario as far as validations and cluster creation. We expect the deploy would fail. If the vc were to be dualstack, it may just work. Either way, the address would be excluded from node IP selection, if the deploy were to succeed. Perhaps there's a separate feature/validation to be queued up to address this scenario.

Reply:
I see. Thanks for the explanation.